File: src/main/java/com/uwetrottmann/trakt5/enums/ListPrivacy.java
package com.uwetrottmann.trakt5.enums;

import com.google.gson.annotations.SerializedName;

public enum ListPrivacy {

    @SerializedName("private")
    PRIVATE,

    @SerializedName("friends")
    FRIENDS,

    @SerializedName("public")
    PUBLIC
}

File: src/main/kotlin/com/redissi/trakt/enums/ListPrivacy.kt
package com.redissi.trakt.enums

import java.util.*

enum class ListPrivacy(val value: String) : TraktEnum {
    PRIVATE("private"), FRIENDS("friends"), PUBLIC("public");

    companion object {
        private val STRING_MAPPING: MutableMap<String, ListPrivacy> =
            HashMap()

        fun fromValue(value: String?): ListPrivacy? {
            return STRING_MAPPING[value]
        }

        init {
            for (via in values()) {
                STRING_MAPPING[via.toString()] = via
            }
        }
    }

    override fun toString(): String {
        return value
    }

}